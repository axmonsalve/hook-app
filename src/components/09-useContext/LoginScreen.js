import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  const obj = {
    id: 123,
    nombre: 'Axel',
  };
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser(obj)}>
        Login
      </button>
    </div>
  );
};
