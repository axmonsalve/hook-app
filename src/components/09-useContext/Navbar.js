import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const Navbar = () => {
  return (
    <nav className="navbar navbar-expand navbar-dark bg-primary mb-3">
      <div className="container-fluid">
        <Link to="/" className="navbar-brand">
          Navbar
        </Link>

        <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
          <div className="navbar-nav">
            <NavLink exact activeClassName="active" to="/" className="nav-link">
              Home
            </NavLink>
            <NavLink
              exact
              activeClassName="active"
              to="/about"
              className="nav-link"
            >
              About
            </NavLink>
            <NavLink
              exact
              activeClassName="active"
              to="/login"
              className="nav-link"
            >
              Login
            </NavLink>
          </div>
        </div>
      </div>
    </nav>
  );
};
