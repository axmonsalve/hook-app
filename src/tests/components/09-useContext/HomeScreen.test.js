import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en el componente HomeScreen', () => {
  const user = {
    name: 'Axel',
    email: 'axel@correo.com',
  };
  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <HomeScreen />
    </UserContext.Provider>
  );
  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
