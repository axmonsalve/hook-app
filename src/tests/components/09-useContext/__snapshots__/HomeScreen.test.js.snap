// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente HomeScreen Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Axel",
   "email": "axel@correo.com"
}
  </pre>
</div>
`;
